rectools/models/candidate_ranking.py

```
@@ -0,0 +1,637 @@
+        for_train : bool
+            Whether the fitting is for training or not.
+        """
+        for candgen in self.cand_gen_dict.values():
+            candgen.fit(dataset, for_train)
+
+    def _get_candidates_from_first_stage(
+        self,
+        users: ExternalIds,
+        dataset: Dataset,
+        filter_viewed: bool,
+        for_train: bool,
+        items_to_recommend: tp.Optional[ExternalIds] = None,
+        on_unsupported_targets: ErrorBehaviour = "raise",
+    ) -> pd.DataFrame:
+        """
+        Get candidates from the first-stage models.
+
+        Parameters
+        ----------
+        users : ExternalIds
+            List of user IDs to get candidates for.
+        dataset : Dataset
+            The dataset to get candidates from.
+        filter_viewed : bool
+            Whether to filter already viewed items.
+        for_train : bool
+            Whether the candidates are for training or not.
+        items_to_recommend : tp.Optional[ExternalIds], optional
+            List of items to recommend. Default is None.
+
+        Returns
+        -------
+        pd.DataFrame
+            DataFrame containing the candidates.
+        """
+        candidates_dfs = []
+
+        for identifier, candgen in self.cand_gen_dict.items():
+            candidates = candgen.generate_candidates(
+                users=users,
+                dataset=dataset,
+                filter_viewed=filter_viewed,
+                for_train=for_train,
+                items_to_recommend=items_to_recommend,
+                on_unsupported_targets=on_unsupported_targets,
+            )
+
+            # Process ranks and scores as features
+            rank_col_name, score_col_name = f"{identifier}_rank", f"{identifier}_score"
+
+            candidates.rename(
+                columns={Columns.Rank: rank_col_name, Columns.Score: score_col_name},
+                inplace=True,
+                errors="ignore",
```

Please add comment here why we ignore errors

---

I don't remember why it's here :)
Removed